### PR TITLE
[ticket/16787] Properly check if fsockopen() is enabled - PHP 8

### PR DESCRIPTION
**Pull request by 3D-I**
PHPBB3-16787

Checklist:

- [x] Correct branch: master for new features; 3.3.x for fixes
- [x] Tests pass
- [x] Code follows coding guidelines: [master](https://area51.phpbb.com/docs/dev/master/development/coding_guidelines.html) and [3.3.x](https://area51.phpbb.com/docs/dev/3.3.x/development/coding_guidelines.html)
- [x] Commit follows commit message [format](https://area51.phpbb.com/docs/dev/3.3.x/development/git.html)

https://tracker.phpbb.com/browse/PHPBB3-16787

### CODE REVIEW

**Comment by 3D-I** [comment]
I'm not sure I understand the problem with the test, as usual.

File: phpBB/phpbb/file_downloader.php [hunk 1]
```diff
@@ -42,7 +42,9 @@ public function get($host, $directory, $filename, $port = 80, $timeout = 6)
 		$this->error_number = 0;
 		$this->error_string = '';
 
-		if ($socket = @fsockopen(($port == 443 ? 'ssl://' : '') . $host, $port, $this->error_number, $this->error_string, $timeout))
+		if (function_exists('fsockopen') &&
+			$socket = fsockopen(($port == 443 ? 'ssl://' : '') . $host, $port, $this->error_number, $this->error_string, $timeout)
```
Comment (marc1706) [inline phpBB/phpbb/file_downloader.php:46]: Add this back:
```suggestion
			$socket = @fsockopen(($port == 443 ? 'ssl://' : '') . $host, $port, $this->error_number, $this->error_string, $timeout)
```

**Review by marc1706 — CHANGES_REQUESTED** [review]

**Comment by rubencm** [comment]
Can that be just replaced by file_get_contents?, is much simpler

**Comment by 3D-I** [comment]
> 
> 
> Can that be just replaced by file_get_contents?, is much simpler
> Edit: Do we really need a class for this?

In a perfect world I would say that the use of different file reading techniques should be used, using code that automatically knows which one to use for each server configuration (cURL, file_get_contents, fsockopen etc...) or something like Guzzle in case. This before having to give a negative response and show an error to the end user. This PR proposes the solution for the problem highlighted by the ticket itself, only. As for the class, I would say that this is a question @marc1706 can answer as he is the one who wrote it 7 years ago.

**Comment by marc1706** [comment]
This class replaces the original get_remote_file() function we had. So yes, moving this to a class was a viable step. Back then Guzzle was not an option yet. It could make sense to move this to use something like Guzzle, however that is outside the scope of this ticket and hence will require a new one.

**Review by marc1706 — APPROVED** [review]

**Comment by 3D-I** [comment]
> 
> 
> ... however that is outside the scope of this ticket and hence will require a new one.

Perhaps this ticket could be used on purpose.
https://tracker.phpbb.com/browse/PHPBB3-16738

**Pull request merged**